avoid that mpi4py initializes MPI at import,
only initialize when needed
cfr. https://github.com/h5py/h5py/pull/1544
author: Kenneth Hoste (HPC-UGent)
diff -ru h5py-2.10.0.orig/h5py/_hl/files.py h5py-2.10.0/h5py/_hl/files.py
--- h5py-2.10.0.orig/h5py/_hl/files.py	2019-09-06 23:29:33.000000000 +0200
+++ h5py-2.10.0/h5py/_hl/files.py	2020-05-12 17:56:43.503885936 +0200
@@ -44,6 +44,8 @@
 
 def _set_fapl_mpio(plist, **kwargs):
     import mpi4py
+    if not mpi4py.MPI.Is_initialized():
+        mpi4py.MPI.Init()
     kwargs.setdefault('info', mpi4py.MPI.Info())
     plist.set_fapl_mpio(**kwargs)
 
diff -ru h5py-2.10.0.orig/h5py/__init__.py h5py-2.10.0/h5py/__init__.py
--- h5py-2.10.0.orig/h5py/__init__.py	2019-09-06 23:33:01.000000000 +0200
+++ h5py-2.10.0/h5py/__init__.py	2020-05-12 17:56:12.603359778 +0200
@@ -16,6 +16,15 @@
 
 from warnings import warn as _warn
 
+# Tell mpi4py to not automatically initialize MPI at import;
+# required to avoid problems with 'import h5py' in some environment
+# when h5py was built with MPI support
+try:
+    import mpi4py
+    mpi4py.rc.initialize = False
+    mpi4py.rc.finalize = True
+except ImportError:
+    pass
 
 # --- Library setup -----------------------------------------------------------
 
